witchcraft/witchcraft.go

```diff
@@ -744,6 +760,11 @@ func (s *Server) initRuntimeConfig(ctx context.Context) (rBaseCfg refreshableBas
 		return cfgBytes
 	})
 
+	unmarshalYAMLFn := yaml.Unmarshal
```

Comment:
can we add `unmarshalYAMLFn` as a field on witchcraft object instead of `strictUnmarshalConfig bool`? and then use that everywhere?

Reply:
Sure. I don't think we should allow it to be customized beyond these values for now (without a more comprehensive change), but since this is fully controlled in code that should be fine.